corehq/project_limits/models.py

```diff
@@ -97,3 +102,39 @@ class Meta:
     def __str__(self):
         domain = f"[{self.domain}] " if self.domain else ""
         return f"{domain}{self.key}: {self.limit}"
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        SystemLimit._get_global_limit.clear(self.key)
+        SystemLimit._get_domain_specific_limit.clear(self.key, self.domain)
+
+    def delete(self, *args, **kwargs):
+        super().delete(*args, **kwargs)
+        SystemLimit._get_global_limit.clear(self.key)
+        SystemLimit._get_domain_specific_limit.clear(self.key, self.domain)
+
+    @classmethod
+    def for_key(cls, key, domain=''):
+        """
+        Return the value associated with the given key, prioritizing the domain specific entry over the general one
+        """
+        domain_limit = cls._get_domain_specific_limit(key, domain) if domain else None
+        if domain_limit is not None:
+            return domain_limit
+        return cls._get_global_limit(key)
+
+    @staticmethod
+    @quickcache(['key'], timeout=FIVE_MINUTES, skip_arg=lambda *args: settings.UNIT_TESTING)
```

Comment:
Is it fine that this caching is scoped to request/celery task? (That's the default behavior of quickcache without a custom `session_function`.)

Aside: I wonder how quickcache works in pillow processes? I assume a it's the same as a single celery task.

Reply:
> Is it fine that this caching is scoped to request/celery task? (That's the default behavior of quickcache without a custom session_function.)

Local caching is scoped to request/task right? As my other comment outlines, I think hitting redis frequently to start is fine, and I can monitor to see if additional changes are needed.